text/0000-cargo-install.md

```diff
@@ -0,0 +1,271 @@
+
+* Dynamic native libraries built as part of `cargo build`.
+* Native assets such as images not included in the binary itself.
+* The source code is not guaranteed to exist, and the binary doesn't know where
+  the source code is.
+
+Additionally, Cargo will not immediately provide the ability to configure the
+installation stage of a package. There is often a desire for a "pre-install
+script" which runs various house-cleaning tasks. This is left as a future
+extension to Cargo.
+
+# Drawbacks
+
+Beyond the standard "this is more surface area" and "this may want to
+aggressively include more features initially" concerns there are no known
+drawbacks at this time.
+
+# Alternatives
+
+### System Package Managers
+
+The primary alternative to putting effort behind `cargo install` it to instead
+put effort behind system-specific package managers. For example the line between
+a system package manager and `cargo install` is a little blurry, and the
+"official" way to distribute a package should in theory be through a system
+package manager. This also has the upside of benefiting those outside the Rust
+community as you don't have to have Cargo installed to manage a program. This
+approach is not without its downsides, however:
+
+* There are *many* system package managers, and it's unclear how much effort it
+  would be for Cargo to support building packages for all of them.
+* Actually preparing a package for being packaged in a system package manager
+  can be quite onerous and is often associated with a high amount of overhead.
+* Even once a system package is created, it must be added to an online
+  repository in one form or another which is often different for each
+  distribution.
+
+All in all, even if Cargo invested effort in facilitating creation of system
+packages, **the threshold for distribution a Rust program is still too high**.
+If everything went according to plan it's just unfortunately inherently complex
+to only distribute packages through a system package manager because of the
+various requirements and how diverse they are. The `cargo install` command
+provides a cross-platform, easy-to-use, if Rust-specific interface to installing
+binaries.
+
+It is expected that all major Rust projects will still invest effort into
+distribution through standard package managers, and Cargo will certainly have
+room to help out with this, but it doesn't obsolete the need for
+`cargo install`.
+
+### Installing Libraries
+
+Another possibility for `cargo install` is to not only be able to install
+binaries, but also libraries. The meaning of this however, is pretty nebulous
+and it's not clear that it's worthwhile. For example all Cargo builds will not
+have access to these libraries (as Cargo retains control over dependencies). It
+may mean that normal invocations of `rustc` have access to these libraries (e.g.
+for small one-off scripts), but it's not clear that this is worthwhile enough to
+support installing libraries yet.
```

Comment:
Unifying multirust and cargo would help dramatically with this. I very often find myself wanting a number of libraries for one of Rust files.